BUILD_OPTIONS.md

```diff
@@ -1,14 +1,25 @@
 # Customizing the Build
 
 **Table of contents**
+- [Rust configuration flags](#rust-configuration-flags)
 - [Features](#features)
 - [sdkconfig](#sdkconfig)
-- [Build configuration](#build-configuration)
-- [Extra esp-idf components](#extra-esp-idf-components)
+- [ESP-IDF configuration](#esp-idf-configuration)
```

Comment:
I still prefer it to be spelled lowercase, less yelling. But I guess that's just me.